content/blog/2022/2022-01-18-getting-started-with-spring-boot.md

```diff
@@ -0,0 +1,915 @@
+#### Field-based Injection
+Let us look at how to use the field-based injection:
+```java
+@Service
+public class GetBookService {
+    @Autowired
+    private final BookRepository bookRepository;
+
+    // Rest of the code is omitted
+}
+```
+
+Spring recognizes the `@Autowired` annotation and makes sure that the `BookRepository` is provided.
+#### Setter-based Injection
+```java
+@Service
+public class GetBookService {
+
+    private BookRepository bookRepository;
+
+    @Autowired
+    private void setBookRepository(BookRepository bookRepository){
+        this.bookRepository = bookRepository;
+    }
+
+    // Rest of the code is omitted
+}
+```
+
+We can put the `@Autowired` annotation on the setter method. The setter-based injection does not allow us to mark the variable as final.
+#### Constructor-based Injection
+```java
+@Service
+public class GetBookService {
+
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public GetBookService(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
+
+    // Rest of the code is omitted
+}
+```
+For the constructor-based annotation, we should set the `@Autowired` annotation on the constructor. 
+
+We are sure that the IoC container will provide all constructor arguments.
+
+More about dependency injection and why to use the constructor-based injection we can find in the [article.](https://reflectoring.io/constructor-injection/)
+
+## Building Controllers
+After creating entities, repositories, and services let us create our first controller that is solving the requirement that the application should be accessable through the browser.
+
+{{% image alt="Spring Boot High-level arch " src="images/posts/spring-boot-begginer-guide/SpringApplication-webLayer.png" %}}
+
+The imported `spring-boot-starter-web` contains everything that we need for the controller. We have Spring MVC autoconfigured and the local server ready to use.
+
+With Spring MVC, we can define a controller with the `@Controller` or `@RestController` annotation so it can handle incoming requests.
+
+### Creating an Endpoint With the @RestController
+We are going to create our first endpoint. We want to fetch all books that the bookstore owns. This is the endpoint that will be called from the homepage.
+
+Let us look into the codebase:
+```java
+@RestController
+@RequestMapping("/books")
+public class BooksRestController {
+
+    private final GetBookService getBookService;
+
+    @Autowired
+    public BooksRestController(GetBookService getBookService) {
+        this.getBookService = getBookService;
+    }
+
+    @GetMapping
+    List<BookResponse> fetchAllBooks(){
+        return getBookService.getAllBooks();
+    }
+
+    // Rest of the code 
+}
+```
+To make our class the controller bean we need to annotate it with `@RestController` or with `@Controller`.
+The difference between these two is that `@RestController` automatically wraps the object into `ResponseEntity<>`. 
```

Comment:
Which object? The return value of all methods that answer to a request, i.e. all methods that are annotated with `@GetMapping` etc.